CodeBuild accessing DockerHub unauthenticated causing build failures

Here's the error

```
Step 1/34 : FROM centos:7
7: Pulling from library/centos
toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
Makefile:31: recipe for target 'build' failed
```

Here's a doc about how to fix this

https://aws.amazon.com/premiumsupport/knowledge-center/codebuild-docker-pull-image-error/

Here's where the codebuild IAM role is configured

https://github.com/mozilla-iam/iam-infra/blob/main/terraform/global/codebuild/sso-dashboard/main.tf

Which makes me think the tasks here are

- [ ] Create a dedicated DockerHub user for SSO Dashboard Codebuild
- [ ] Save those credentials wherever your team saves those kinds of things
- [ ] Save them into Parameter Store because the existing CodeBuild IAM Role permits access to the `/iam/sso-dashboard/` path in parameter store
- [ ] Update the Makefile so the `setup-codebuild` section authenticates to DockerHub

Downside to this approach is that it only solves for SSO dashboard, whereas all CodeBuild docker driven jobs will have this problem.

To solve more generally you'd need to create a shared parameter store location where multiple apps could access the dockerhub credentials and modify applications codebuild IAM roles (in terraform) to grant access to those paths in parameter store.

--- a/Makefile
+++ b/Makefile
@@ -23,6 +23,9 @@ setup-codebuild:
 
 .PHONY: login
 login:
+	$(eval DOCKER_ID := $(shell aws --region us-west-2 ssm get-parameter --name "/iam/sso-dashboard/$(STAGE)/docker_user_id" --query 'Parameter.Value' --output text))
+	$(eval DOCKER_ACCESS_TOKEN := $(shell aws --region us-west-2 ssm get-parameter --name "/iam/sso-dashboard/$(STAGE)/docker_access_token" --query 'Parameter.Value' --output text))
+	echo $(DOCKER_ACCESS_TOKEN) | docker login --username $(DOCKER_ID) --password-stdin
 	aws --region us-west-2 eks update-kubeconfig --name $(CLUSTER_NAME)
 	aws ecr get-login --region us-west-2 --no-include-email | bash
 

--- a/Makefile
+++ b/Makefile
@@ -24,7 +24,7 @@ setup-codebuild:
 .PHONY: login
 login:
 	$(eval DOCKER_ID := $(shell aws --region us-west-2 ssm get-parameter --name "/iam/sso-dashboard/$(STAGE)/docker_user_id" --query 'Parameter.Value' --output text))
-	$(eval DOCKER_ACCESS_TOKEN := $(shell aws --region us-west-2 ssm get-parameter --name "/iam/sso-dashboard/$(STAGE)/docker_access_token" --query 'Parameter.Value' --output text))
+	$(eval DOCKER_ACCESS_TOKEN := $(shell aws --region us-west-2 ssm get-parameter --name "/iam/sso-dashboard/$(STAGE)/docker_access_token" --with-decryption --query 'Parameter.Value' --output text))
 	echo $(DOCKER_ACCESS_TOKEN) | docker login --username $(DOCKER_ID) --password-stdin
 	aws --region us-west-2 eks update-kubeconfig --name $(CLUSTER_NAME)
 	aws ecr get-login --region us-west-2 --no-include-email | bash

